dot/sync/unready_blocks.go

```diff
@@ -36,34 +101,32 @@ func (u *unreadyBlocks) newIncompleteBlock(blockHeader *types.Header) {
 	}
 }
 
-func (u *unreadyBlocks) newDisjointFragment(frag []*types.BlockData) {
+func (u *unreadyBlocks) newDisjointFragment(frag *Fragment) {
 	u.mtx.Lock()
 	defer u.mtx.Unlock()
 	u.disjointFragments = append(u.disjointFragments, frag)
 }
 
 // updateDisjointFragments given a set of blocks check if it
-// connects to a disjoint fragment, if so we remove the fragment from the
-// disjoint set and return the fragment concatenated with the chain argument
-func (u *unreadyBlocks) updateDisjointFragments(chain []*types.BlockData) ([]*types.BlockData, bool) {
+// connects to a disjoint fragment, and returns a ne fragment
+func (u *unreadyBlocks) updateDisjointFragment(chain *Fragment) (*Fragment, bool) {
```

Comment:
```suggestion
func (u *unreadyBlocks) updateDisjointFragments(chain *Fragment) (*Fragment, bool) {
```
I would keep the plural here. Alternatively, update the doc comment as well.